src/tqec/block/block.py

```diff
@@ -327,17 +332,21 @@ def instantiate_without_boundary(self, boundary: Direction3D) -> cirq.Circuit:
     def scale_to(self, k: int) -> None:
         self.template.scale_to(k)
 
-    def _update_plaquettes(self, sides: list[TemplateSide]) -> None:
+    def _update_plaquettes(
+        self, sides: list[TemplateSide], replacement: StandardComputationBlock
+    ) -> None:
         "Replaces plaquettes according to substitution rules."
         for index in self.template.get_plaquette_indices_on_sides(sides):
-            replacement_index = self.template.get_corresponding_mid_for_side(index)
-            self.plaquettes.initial_plaquettes[index] = (
-                self.plaquettes.initial_plaquettes[replacement_index]
-            )
-            self.plaquettes.final_plaquettes[index] = self.plaquettes.final_plaquettes[
-                replacement_index
+            self.plaquettes.initial_plaquettes[index] = replacement.initial_plaquettes[
+                index
+            ]
+            self.plaquettes.final_plaquettes[index] = replacement.initial_plaquettes[
+                index
```

Comment:
@nelimee this is how I would use the computation blocks from the pipes.
This works based on the assumption that the replacement block has the same indices on the border as the original block, which is not true at the moment.
Do you think it is possible to have `QubitHorizontalBorders` look like
```
1  5  6  5  6 2
3 13 14 13 14 4
```
instead?